android/app/src/main/java/com/zulipmobile/MainActivity.kt

```diff
@@ -108,26 +108,35 @@ open class MainActivity : ReactActivity() {
         // For documentation of what fields to expect here, see:
         //   https://developer.android.com/reference/android/content/Intent#ACTION_SEND
         val params = Arguments.createMap()
-        when {
-            "text/plain" == intent.type -> {
-                val sharedText = intent.getStringExtra(Intent.EXTRA_TEXT)
-                params.putString("type", "text")
-                params.putString("sharedText", sharedText)
-            }
-            intent.type?.startsWith("image/") == true -> {
```

Comment:
`contentResolver.getType` has the potential to return null, so in the new revision I am handling this in kotlin code itself although I am not sure if this is the right way, it returns null only when the url provided is broken, this shouldn't really happen with SEND_INTENT but just in case it does, I give a fallback mimeType for `octet-stream`.

implementation in master branch deduces types by directly analyzing the send intent, but this won't be possible in case of multiple files, it will just be */* in that case, wonder what will happen there if a broken url is provided.